openwisp_network_topology/api/serializers.py

```diff
@@ -22,15 +31,193 @@ def to_representation(self, data):
         )
 
 
-class NetworkGraphSerializer(serializers.ModelSerializer):
+def get_receive_url(pk, key):
+    path = reverse('receive_topology', args=[pk])
+    url = '{0}?key={1}'.format(path, key)
+    return url
+
+
+def get_representation_data(obj):
     """
-    NetJSON NetworkGraph
+    Returns a dict that represents
+    a NetJSON NetworkGraph object.
     """
+    nodes = []
+    links = []
+    for link in obj.get_links_queryset():
+        links.append(link.json(dict=True, original=False))
+    for node in obj.get_nodes_queryset():
+        nodes.append(node.json(dict=True, original=False))
+    netjson = OrderedDict(
+        (
+            ('type', 'NetworkGraph'),
+            ('protocol', obj.protocol),
+            ('version', obj.version),
+            ('metric', obj.metric),
+            ('label', obj.label),
+            ('id', str(obj.id)),
+            ('parser', obj.parser),
+            ('organization', str(obj.organization.id)),
+            ('strategy', obj.strategy),
+            ('url', obj.url),
+            ('key', obj.key),
+            ('expiration_time', obj.expiration_time),
+            ('receive_url', get_receive_url(obj.pk, obj.key)),
+            ('published', obj.published),
+            ('created', obj.created),
+            ('modified', obj.modified),
+            ('nodes', nodes),
+            ('links', links),
+        )
+    )
+    return netjson
+
 
+class NetworkGraphRepresentation(object):
     def to_representation(self, obj):
-        return obj.json(dict=True)
+        """
+        Overriding the default represenation
+        of Topology object.
+        """
+        topo_data = get_representation_data(obj)
```

Comment:
Can we please have proper variable names, `topo_data` -> `topology_data`! :smile:

Reply:
ok, Done